using JSR 356 for websockets

alexsaveliev opened the pull request:
@birdowl PTAL and play a bit with my changes, I may have forgot something or did something wrong.
I have tested my changes with Jetty 9.4 and Tomcat 8.5 but they supposed to work with any JSR 356 compliant container

alexsaveliev committed:
WIP

- s/jetty/websocket
- removed jetty dependencies
- renamed classes

alexsaveliev committed:
dependencies and submodules reorganization

- moved all artifacts to the same group, renamed some
- moved common dependencies to parent module

alexsaveliev committed:
make it compile

alexsaveliev committed:
switching to Maven snapshots

- s/1.0.2/1.0.2-SNAPSHOT
- changed packaging of sample/chat from jar to war

alexsaveliev committed:
packaging sample/chat as webapp

changed directory layout to produce webapp

alexsaveliev committed:
first working version

- made `SocketIOManager` singleton because it must be shared between servlets and websocket endpoints
- storing servlet config in a singleton holder to be able to reuse it by websocket endpoints
- demo doesn't use `socket.getRequest` anymore because request information is missing in websocket mode

alexsaveliev committed:
removed commented out block

birdowl reviewed (dismissed):
Great job! It looks like we do not need 'extension' module anymore. Let's simplify the structure and move everything into one module. We do not need to name it 'core' anymore since we do not anticipate having extensions

alexsaveliev committed:
merged core and extension modules into one

alexsaveliev committed:
s/core/socket-io

alexsaveliev committed:
embedded Jetty server sample

- added sample that launches embedded Jetty server with socket.io chat application

alexsaveliev committed:
embedded Tomcat server example (WIP)

- embedded Tomcat server example
- dealing with jar vs directories packaging  when identifying resource base
- added `onError` stub

alexsaveliev committed:
corrections

- removed Jasper from packaging because it clashes with websocket (overrides `META-INF/services/javax.servlet.ServletContainerInitializer`)

alexsaveliev committed:
updated README
